libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -169,14 +164,14 @@ void AC_WPNav::wp_and_spline_init(float speed_cms, Vector3f stopping_point)
     _wp_desired_speed_xy_cms = MAX(_wp_desired_speed_xy_cms, WPNAV_WP_SPEED_MIN);
 
     // initialise position controller speed and acceleration
-    _pos_control.set_max_speed_accel_xy(_wp_desired_speed_xy_cms, _wp_accel_cmss);
-    _pos_control.set_correction_speed_accel_xy(_wp_desired_speed_xy_cms, _wp_accel_cmss);
+    _pos_control.set_max_speed_accel_xy(_wp_desired_speed_xy_cms, get_wp_acceleration());
+    _pos_control.set_correction_speed_accel_xy(_wp_desired_speed_xy_cms, get_wp_acceleration());
     _pos_control.set_max_speed_accel_z(-get_default_speed_down(), _wp_speed_up_cms, _wp_accel_z_cmss);
     _pos_control.set_correction_speed_accel_z(-get_default_speed_down(), _wp_speed_up_cms, _wp_accel_z_cmss);
 
     // calculate scurve jerk and jerk time
     if (!is_positive(_wp_jerk)) {
-        _wp_jerk = _wp_accel_cmss;
+        _wp_jerk.set(get_wp_acceleration());
```

Comment:
this isn't a change in behaviour, but it looks wrong. I think it means the WP_JERK param value used depends on the geometry of the first turn.
it would be better to not change the parameter and calculate the value we want to use instead?

Reply:
As you say this is not a change in behaviour. This is set when jerk to an impossible number, ie zero or negative.
The relationship between maximum jerk, acceleration, velocity, waypoint delay, waypoint radius and corner angle define the resultant corner geometry. 

This value can't be right or wrong, only possible or impossible. calc_scurve_jerk_and_snap() adjusts this value further if needed. The value set here will tend to give a reasonable corner geometry so is at least as good a setting as any other.

Reply:
then why not set a fixed default? We do that for invalid parameters in many places. That at least gives predictable behaviour on each flight. Doing it this way gives a different value on each flight

Reply:
discussed, looks fine, thanks!